src/app/content/highlights/components/CardWrapper.tsx

```diff
@@ -11,20 +15,109 @@ interface Props {
   className?: string;
 }
 
+const displayAllCardsMediaQuery = `(max-width: ${mediaQueryBreakToStopDisplaingAllCards})`;
+
 // tslint:disable-next-line:variable-name
-const Wrapper = ({highlights, className, container, highlighter}: Props) => highlights.length
-  ? <div className={className}>
+const Wrapper = ({highlights, className, container, highlighter}: Props) => {
+  const displayAllCards = !assertWindow().matchMedia(displayAllCardsMediaQuery).matches;
+
+  const element = React.useRef<HTMLElement>(null);
+  const [cardsPositions, setCardsPositions] = React.useState<Map<string, number>>(new Map());
+  const [cardsHeights, setCardsHeights] = React.useState<Map<string, number>>(new Map());
+
+  const onHeightChange = (id: string, ref: React.RefObject<HTMLElement>) => {
+    if (!displayAllCards) { return; }
```

Comment:
> do you really need to short circuit these functions?

nope

> isn't it fine for them to update anyway since the other cards are all hidden by css?

I would say no because it would still consume cpu power and not provide any benefit for the user

Reply:
but in order to actually tell if the cards are all shown, your logic will have to become much more complicated to compensate for the different breakpoints for each sidebar's state. i think it might be worth the extra (probably nominal) processing to avoid this complexity.